articles/ros2_security_contexts.md

```diff
@@ -0,0 +1,307 @@
+---
+layout: default
+title: ROS 2 Security Contexts
+permalink: articles/ros2_security_contexts.html
+abstract:
+  This article specifies the integration between security and contexts.
+author:  >
+  [Ruffin White](https://github.com/ruffsl),
+  [Mikael Arguedas](https://github.com/mikaelarguedas)
+
+published: false
+categories: Security
+---
+
+{:toc}
+
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+
+TODO: Some concise overview introduction here
+
+## Concepts
+
+Before detailing the SROS 2 integration of the contexts, the following concepts are introduced.
+
+### Participant
+
+Participant is the object representing a single entity on the network, in the case of DDS the ``Participant`` is a DDS DomainParticipant and the object to which a set of access control and security identity apply.
+
+### Namespaces
+
+Namespaces are a fundamental design pattern in ROS and are widely used to organize and differentiate many types of resources as to be uniquely identifiable; i.e. for topics, services, actions, and node names.
+As such, the concept of namespacing is well know and understood by current users, as well as strongly supported with the existing tooling.
+Namespaces are often configurable at runtime via command line arguments or statically/programmatically via launch file declarations.
+
+Previously, the Fully Qualified Name (FQN) of a node was used directly by a selected security directory lookup strategy to load the necessary key material.
+However, with the advent of contexts, such a direct mapping of FQN to security artifacts may no longer suffice.
+
+### Contexts
+
+With the advent of ROS 2, multiple nodes may now be composed into one process for improved performance.
+Previously however, each node would retain it's one to one mapping to a separate middleware ``Participant``.
+Given the non-negligible overhead incurred of multiple ``Participant``s per process, a concept of contexts was introduced.
+Contexts permit a many-to-one mapping of nodes to ``Participant`` by grouping many nodes per context, and one ``Participant`` per context.
```

Comment:
This will need a slight rephrase as well, to reflect that contexts already existed.

```suggestion
Given the non-negligible overhead incurred of multiple ``Participant``s per process, a change was introduced to map a single ``Participant`` to a context, and allow for multiple nodes to share that context.
```